### ISSUE TITLE
Unable to find linked item at path

### ISSUE THREAD

**Issue by ferferga** [jellyfin/jellyfin#1907]
**Describe the bug**
Changing the filenames or path of some items added in a collection results in "orphaned" items in the collection. This error is represented in the logs at Warning level like this:
``[WRN] Unable to find linked item at path path/to/the/original/file``

**To Reproduce**
1. Add items to a collection
2. Change the path or name of those items in your filesystem
3. Do an scan. Items will be automatically identified if you didn't change the name schema in a way that isn't recognizable by Jellyfin. If that's the case, Jellyfin will add the items to the original collection again.
4. Reboot the server and scan again (optional, but to make sure that Jellyfin is fully "aware" of the change)
5. See the logs after that last scan and grep with the filename of the old file: You will see that, although it handled the filename change gracefully in the first scan, it's still expecting the old file to be there, as that error just keeps spawning.

No matter how many library scans you do afterwards, those files will be stuck forever and Jellyfin will always be expecting them, although the user interface in all apps show the collection pointing to the new path correctly.

**Expected behavior**
Jellyfin forgets everything about all the old files in collections

**System (please complete the following information):**
 - OS: Debian Buster in Raspberry Pi
 - Jellyfin Version: [e.g. 10.4.0]
 - Reverse proxy: nginx

**Additional context**

I haven't tested this, but I think that probably this is happening if you just remove an item from the filesystem, as Jellyfin, when a file is altered/moved from the filesystem, removes it and adds it again.

**Comment by anthonylavado** [comment]
In the case of music, this error can come up due to https://github.com/jellyfin/jellyfin/issues/1874

**Comment by ferferga** [comment]
@anthonylavado In my case, it happened only with movies

**Comment by snis** [comment]
Exakt same issue with removed media 
`[WRN] Unable to find linked item at path ...`

- OS: Arch linux x64
- Jellyfin v 10.4.3

**Comment by Deepcuts** [comment]
Same problem here on 10.5.2

**Comment by manzajaro** [comment]
> Same problem here on 10.5.2

me too,

OS: Linux  Arm64
Jellyfin v 10.5.2

**Comment by dkanada** [comment]
I have looked into the scanning before and we have helper classes that should clean missing files and such from the database. I was looking at the album artist one but there might be one for other objects as well.

**Comment by cognicom** [comment]
v10.5.3 on Ubuntu (Debian) still has the problem. I don't store any music on the server, but have been replacing old .AVI files from DVD with .MP4 and .MKV files from BluRay, and each day's log file is larger with ever-more "unable to find linked item" entries (referring to the old .AVI files).

Apart from the ongoing logfile growth, it seems to be impacting the time taken to scan the library as it chugs through looking for the defunct files.

Example:
[2020-04-18 00:26:50.192] [WRN] Unable to find linked item at path "/mnt/videos/Movies/Science Fiction/Star Wars 5 - The Empire Strikes Back (1980).avi"
[2020-04-18 00:26:50.194] [WRN] Unable to find linked item at path "/mnt/videos/Movies/Science Fiction/Star Wars 6 - Return of the Jedi (1983).avi"
[2020-04-18 00:26:50.194] [WRN] Unable to find linked item at path "/mnt/videos/Movies/Science Fiction/Star Wars 4 - A New Hope (1977).avi"
[2020-04-18 00:26:50.195] [WRN] Unable to find linked item at path "/mnt/videos/Movies/Science Fiction/Star Wars 1 - The Phantom Menace (1999).avi"
[2020-04-18 00:26:50.195] [WRN] Unable to find linked item at path "/mnt/videos/Movies/Science Fiction/Star Wars 2 - Attack of the Clones (2002).avi"
[2020-04-18 00:26:50.196] [WRN] Unable to find linked item at path "/mnt/videos/Movies/Science Fiction/Star Wars 3 - Revenge of the Sith (2005).avi"

**Comment by cognicom** [comment]
So, to follow up on this... I don't know about the issue with music files, but my issue with video files is attributable to the "TMDb Box Sets" plug-in. Removing the plug-in and deleting all collections created by it has yielded a log file bereft of the aforementioned warning entries.

I suspect that the issue still may be inside Jellyfin itself though, as it's highly likely that had I not deleted all of the collections, the problem would continue. Worth investigating in both.

**Comment by ferferga** [comment]
This is still an issue as of 10.7 unstable

**Comment by oitdmser** [comment]
I can confirm as mentioned earlier that renaming the folder that contains a missing file, running scan library scheduled task, renaming it back to the original name and running scan library again will get Jellyfin to delete the entry. Unfortunately this doesn't work for files that were in the root folder of the library.

**Comment by oitdmser** [comment]
This bug has additional ramifications: Movies that have had their files changed as documented here often cannot be successfully added to collections or playlists. The UI gives no error, they simply fail to appear in the collection. This is likely related to the server attempting and failing to reference the old file that is no longer found.

**Comment by kellcomnet2** [comment]
I can confirm this issue still exists in 10.7 rc 2

**Comment by Hukuma1** [comment]
In RC 3 as well. I guess the only solution right now is to delete and rebuild entire the library?

**Comment by ferferga** [comment]
@Hukuma1 it's spamming the logs but, functionality-wise, nothing is wrong with your library.

It's likely it will appear with your new library as well. Just let it be until the DB is rewritten.

**Comment by Hukuma1** [comment]
It actually didn't even fix it after a full rebuild of the library. And it prevents from movies being added to a collection for me, not just showing errors in logs. :/

I still don't have much, so starting from scratch is easy for now. 

Is the DB re-write coming with 10.7 or later?

**Comment by ferferga** [comment]
@Hukuma1 that's another different issue that warrants its own issue.

Collect as many logs as possible, describe the details as thorough as possible and open a new issue, linking to this one please.

**Comment by Hukuma1** [comment]
I think mine is similar to this: https://github.com/jellyfin/jellyfin/issues/4601

I moved a few movies from directories and thought it would update the paths on its own. (I was wrong haha)

**Comment by GitJamz** [comment]
I have the same issue...
Can not "scan all libraries" because it basically makes no progress after 70 something %.. And in logs every single movie is just bunch of these errors.
But playback and everything on the items works.

I did however do something special.. I copied my data folder from /etc/jellyfin, /var/lib/jellyfin and cache+metadata folder - from  ubuntu 18.x server to Ubuntu 20.x server + update from jellyfin 0.6.4 to 0.7

**Comment by brando56894** [comment]
Still occurs in 10.7.5, I believe this is caused by the TMDB Box Sets plugin.

**Comment by kwm1800** [comment]
I also suffer this issue. Did not know about it until I encountered the error that I could not add a movie to collection.

The logs from docker console are... well, huge. This will be problem as my library grows.

**Comment by rasmuslos** [comment]
If all your collections are automatically generated by the TMDB Box Sets plugin you can delete all of them manually and execute the task "Scan library for missing box sets".

**Comment by ferferga** [comment]
@brando56894 I had it before having that plugin. Plugins can't cause issues with your database, as it's all managed by "core" jellyfin itself. 

I traced down this problem already and it *shouldn't* cause any issue besides spamming the logs and the I/O time it takes for checking the path, so all the people having issues with scanning and so on **it's probably for another cause**, so it's better if you open a new issue (referencing this one anyway so we can trace it down better).

I still need to investigate if this can be fixed before the database is rewritten, but if isn't, this might take quite some time (until 11.x is released)

**Comment by sambartik** [comment]
@ferferga Hello, have you ever resolved this issue?

**Comment by ferferga** [comment]
@sambartik No, this is probably going to keep happening until the library DB rewrite.

**Comment by dschrempf** [comment]
FYI, I also bumped into this (Music files, not Video).

**Comment by michaelkrieger** [comment]
If someone wants to clean up these errors, shutdown Jellyfin, and just remove the movies from the XML.  Then restart Jellyfin and no more errors.  This isn't stored in the database.

I attach a brief PHP script that will do it using the DOM, but if you have a small number of these errors, you can do it manually.  The file was renamed to .txt for ease of upload.

[clean-removed.txt](https://github.com/jellyfin/jellyfin/files/8664740/clean-removed.txt)

**Comment by dschrempf** [comment]
Thank you for this script and the workaround. I still think the old elements should be cleaned up by the server.

**Comment by oitdmser** [comment]
Ran the script from the command line using php and even sudo-ing to ensure it has permissions. The script runs fine, finds the missing items
`Processing Alien v Predator [boxset]/collection.xml...
	/mnt/plex/movies/Prometheus/Prometheus (2012) {imdb-tt1446714}.mp4 doesn't exist.
	Saving Alien v Predator [boxset]/collection.xml... wrote 2406 bytes
`

and if I run it again no missing files are found
`Processing Alien v Predator [boxset]/collection.xml...
`

but when I restart Jellyfin and load the collection from the web interface the same errors appear in the log:
`[2022-05-13 10:46:51.506 -07:00] [WRN] Unable to find linked item at path "/mnt/plex/movies/Prometheus/Prometheus (2012) {imdb-tt1446714}.mp4"`

Note that I am able to add that title to a collection, I just see the errors in the log

**Comment by mdPlusPlus** [comment]
@michaelkrieger I actually had to run the script multiple times until it didn't show any missing files anymore.

**Comment by JPVenson** [comment]
@michaelkrieger same here. I currently have the exact issue and thanks to skribe in discord i found this issue. But i dont like php so i rewrote your skript in c#. https://gist.github.com/JPVenson/e765d97c3c1eb92fb11f0856c6034e6a 
Planning to add it later as a PR in the cleanup task

**Comment by CubeyGlyph** [comment]
Bumping this as I am experiencing this as well.  Looks like they have a task to fix this soon...

**Comment by hugalafutro** [comment]
How would I go about running the script inside a jellyfin docker container? There ain't no php inside can I just install it with apt and then run the script?

**Comment by JPVenson** [comment]
> How would I go about running the script inside a jellyfin docker container? There ain't no php inside can I just install it with apt and then run the script?

You would not run this in the container itself but on the host system and targeting the Mounted config folder as well as the Media folder.
My script has an extra parameter if you want to run it on a different host and need to remap the path root called: `targetPathSubsitution=SOURCEROOT:NEWROOT` so you can replace the root portion of the expected path with a new root for existence check.

**Comment by hugalafutro** [comment]
My Movies library includes 2 sources `/media/movies` and `/media2/movies2`, should I then run it against each of the paths? Just wondering if it can't cause issues like removing all from media2 when ran on media or vice versa.

**Comment by JPVenson** [comment]
@hugalafutro hmm that _could_ work. Try backuping your collections folder and set `/` as the mediaSource path. Also use the `-whatif=y` flag for testing

**Comment by hugalafutro** [comment]
So should I only run it against 1 path? What about the `/media2/movies2`? It's 2 different bind mounts in the docker-compose.yml
```
      - /shares/samba/media:/media
      - /shares/samba/media2:/media2
```
and they actually used to be on 2 different NAS-es, which I since merged to 1 large capacity NAS, but because I didn't want to mess up the library I left the paths as if it still were 2 different mounts.

also, for example in this case:
```
05/03/2023 3:33:56
[03:33:56] [WRN] [26] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path /media/anime/movies/Vampire Hunter D Bloodlust (2000)/Vampire Hunter D Bloodlust (2000) Bluray-1080p HEVC.mkv
05/03/2023 3:33:56
[03:33:56] [WRN] [26] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path /media2/anime/movies/Vampire Hunter D (1985)/Vampire Hunter D (1985) WEBDL-1080p x265.mkv
```
there is no more `/media/anime` only `/media2/anime` so I assume the script would remove both references as the library now only scans one directory, but I also have cases for the Movies library where both of the paths exist, but in both cases I would still run the script only against 1 path? I'm getting mighty confused

**Comment by JPVenson** [comment]
@hugalafutro to keep it short: No the script cannot handle this. If you run it in your scenario, it will remove all collection entries of the other path as it cannot find it.

**Comment by VigibotDev** [comment]
I solved this by removing all Box Sets plugin data inside "/var/lib/jellyfin/data/collections/*" (just delete all file and do a plugin collection rescan)

**Comment by mathsyx-fr** [comment]
I solved the problem by edit each collections.xml in "/var/lib/jellyfin/data/collections/*" and delete the olds paths.

**Comment by widhalmt** [comment]
I have a library with a lots of collections. Many of them manually maintained. So it's not an option for me to just delete them all. I just wanted to let you know, it would be nice, if jellyfin would eventually be able to solve it by itself. No hurries on my side, though.

**Comment by JPVenson** [comment]
> I have a library with a lots of collections. Many of them manually maintained. So it's not an option for me to just delete them all. I just wanted to let you know, it would be nice, if jellyfin would eventually be able to solve it by itself. No hurries on my side, though.

The PR#9231 does fix this.

**Comment by dschrempf** [comment]
#9231 was merged in June. Can we close this? Or is it still an issue?

**Comment by JPVenson** [comment]
I am fine with closing this and opening a new issue if problems with the solution arises
@ferferga

**Comment by wsw70** [comment]
> Or is it still an issue?

It is on version 10.8.12 running in docker via `jellyfin/jellyfin:latest`. I noticed this in my logs after restarting but it does not seem to impact anything.

**Comment by JPVenson** [comment]
@wsw70 The PR is not yet released

**Comment by zotabee** [comment]
> It is on version 10.8.12

Yep, still happening on last version as of today.

> I solved the problem by edit each collections.xml in "/var/lib/jellyfin/data/collections/*" and delete the olds paths.

I confirm this file seems to be the culprit, as it contains old paths for some reasons.

**Comment by ferferga** [comment]
@wsw70 @zotabee This is already fixed in master, the version that's going to get the fix will be from the 10.9.x release cycle.

**Comment by wsw70** [comment]
> Actually you need to be sure to delete the collection from jellyfin UI as it stores some other extra information, not only in the xml file. It worked after that for me.

Thanks. I removed them from the UI, restarted Jellyfin and there are no more errors in the log.

There are no more collections either :) but I did not use them anyway so this is not a loss for me (and they may miraculously get back someday, you never know :))

**Comment by silverwolf04** [comment]
There's something broken with collections even with Version: 10.8.13 . I try adding a movie that has 2 versions to a collection and it refuses to display in the collection (no errors in the log). I deleted all of the collections and tried re-adding it from scratch and the same results occur.

**Comment by michael-zock** [comment]
Still happens to me, even in 10.8.13.
I've tried to delete the auto-generated collections and nothing improves. Made one library practically inaccessible and I'll probably have to delete and recreate the entry to get to those folders again, in the hopes that at least that will help.

**Comment by zotabee** [comment]
Still happening on 10.9.0 apparently.

**Comment by ferferga** [comment]
@zotabee Did your first scan completely successfully? Did you attempt to remove anything manually?

**Comment by zotabee** [comment]
I removed one collection from jellyfin interface as it was "empty" then started a "Scan All Libraries" from the dashboard so it could re-add this particular collection.

**Comment by ferferga** [comment]
@zotabee Don't remove anything. Also relevant: https://matrix.to/#/!NaLbfexyNIBQDbzGwB:bonifacelabs.ca/$lsQY8gZg2BzX9Aet0JSXe8tIykOqECys-9h92zwSFpQ?via=bonifacelabs.ca&via=t2bot.io&via=matrix.org

**Comment by zotabee** [comment]
`<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>R</ContentRating>
  <Added>11/19/2023 16:15:22</Added>
  <LockData>true</LockData>
  <LocalTitle>[REC] Collection</LocalTitle>
  <PremiereDate>2007-11-23</PremiereDate>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <ProductionYear>2007</ProductionYear>
  <Genres>
    <Genre>Horror</Genre>
    <Genre>Mystery</Genre>
    <Genre>Thriller</Genre>
  </Genres>
  <Studios>
    <Studio>Filmax</Studio>
    <Studio>Castelao Productions</Studio>
    <Studio>Canal+ España</Studio>
    <Studio>TVE</Studio>
    <Studio>ICO</Studio>
    <Studio>ONO</Studio>
    <Studio>Rec Apocalypse AIE</Studio>
  </Studios>
  <CollectionItems>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC] (2007) [tmdbid-8329]\Rec2007.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]² (2009) [tmdbid-10664]\Rec2009.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]³ Genesis (2012) [tmdbid-80280]\Rec2010.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]⁴ Apocalypse (2014) [tmdbid-185341]\Rec2014.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC] (2007) [tmdbid-8329]\Rec2007.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]² (2009) [tmdbid-10664]\Rec2009.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]³ Genesis (2012) [tmdbid-80280]\Rec2010.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]⁴ Apocalypse (2014) [tmdbid-185341]\Rec2014.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC] (2007) [tmdbid-8329]\Rec2007.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]² (2009) [tmdbid-10664]\Rec2009.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]³ Genesis (2012) [tmdbid-80280]\Rec2010.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>[REC] Collection [tmdbid-74508]\[REC]⁴ Apocalypse (2014) [tmdbid-185341]\Rec2014.mkv</Path>
    </CollectionItem>
  </CollectionItems>
</Item>`

**Comment by michael-zock** [comment]
9.1 fixed it for me.
There was one errant one that didn't get removed when all the cleanups were through, but that was just a matter or removing the line from the XML manually. Certainly beats several pages worth of error messages wasting resources whenever I'd open some libraries again.

**Comment by GHMJT** [comment]
Still happening jellyfin/jellyfin docker 10.9.9

**Comment by JPVenson** [comment]
@GHMJT you have to get way more detailed for that comment to be helpful.

**Comment by GHMJT** [comment]
I'm looking into it and suspect it might just be my particular installation because there's also other things going on, like collections all being marked as played and not responding to unmarking them.

I did a restore and the issue had not returned. Over the years I've moved source paths many times for various reasons such as consolidating disks, and data with the old paths collectively remain dormant in the database. So the database has accrued a lot of baggage. It's time to start a fresh new library and rescan.

Thank you.

**Comment by michael-zock** [comment]
An old "friend" is back in 10.9, even in the latest 10.9.11 on Win 11 Pro:

```
[2024-09-08 13:18:11.995 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "H:\Video_other\J\The Irresponsible Captain Tylor - Specials (1993) [tmdbid-34189] [imdbid-tt1330612]\S01\The Irresponsible Captain Tylor - S01E03.mkv"
[2024-09-08 13:18:12.000 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "H:\Video_other\J\The Irresponsible Captain Tylor - Specials (1993) [tmdbid-34189] [imdbid-tt1330612]\S01\The Irresponsible Captain Tylor - S01E05.mkv"
[2024-09-08 13:18:12.002 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "H:\Video_other\J\The Irresponsible Captain Tylor - Specials (1993) [tmdbid-34189] [imdbid-tt1330612]\S01\The Irresponsible Captain Tylor - S01E08.mkv"
[2024-09-08 13:18:12.242 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "F:\Video_other\J\Space Runaway Ideon\Space Runaway Ideon - 39 END (BDRip 1440x1080p x265 HEVC FLAC 2.0)[sxales].mkv"
[2024-09-08 13:18:13.342 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "G:\Video_other\J\New Initial D the Movie - Legend 2 - Racer (2015) [tmdbid-321487]\New Initial D the Movie - Legend 2 - Racer -.mkv"
[2024-09-08 13:18:13.344 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "G:\Video_other\J\New Initial D the Movie - Legend 3 - Dream (2016) [tmdbid-372758]\New Initial D the Movie - Legend 3 - Dream -.mkv"
[2024-09-08 13:18:13.346 +02:00] [WRN] [22] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "G:\Video_other\J\New Initial D the Movie - Legend 1 - Awakening (2014) [tmdbid-271709]\New Initial D the Movie - Legend 1 - Awakening - - In 2014, a new legend starts running.mkv"
```
 The collection cleanup task doesn't seem to care either, even though that seemed to fix it when 10.9 was first released.

**Comment by ferferga** [comment]
@michael-zock Can you provide more information about what exactly happened to those files? (moved them elsewhere, deleted, added to another collection, etc...)

**Comment by ferferga** [comment]
@sym0nd0 Can you try in latest unstable? Create a copy of your data and library directories and mount them accordingly in the Docker container, so you run your production instance alongside the testing one.

**Comment by alexvrv** [comment]
Still happening on latest version ... any solution?

**Comment by ShadowWhisperer** [comment]
@alexvrv 

Doing it on mine also, latest version (10.11.4). Removed the files and re-added them. Same issue,

```
[2025-12-03 13:26:25.637 -06:00] [WRN] [17] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "E:\Media\Movies\All Dogs Go To Heaven [1989]\All Dogs Go To Heaven [1989].mkv"
[2025-12-03 13:26:25.638 -06:00] [WRN] [17] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "E:\Media\Movies\Beauty And The Beast [1991]\Beauty And The Beast [1991].mkv"
[2025-12-03 13:26:25.639 -06:00] [WRN] [17] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "E:\Media\Movies\Clear And Present Danger [1994]\Clear And Present Danger [1994].mkv"
[2025-12-03 13:26:32.179 -06:00] [WRN] [10] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "E:\Media\Movies\All Dogs Go To Heaven [1989]\All Dogs Go To Heaven [1989].mkv"
[2025-12-03 13:26:32.180 -06:00] [WRN] [10] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "E:\Media\Movies\Beauty And The Beast [1991]\Beauty And The Beast [1991].mkv"
[2025-12-03 13:26:32.182 -06:00] [WRN] [10] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path "E:\Media\Movies\Clear And Present Danger [1994]\Clear And Present Danger [1994].mkv"

```

**Comment by alexvrv** [comment]
@ShadowWhisperer 

The only solution I found was to use DataGrip, open the jellyfin.db and delete all the lines from BaseItem table that contains paths that doesn't exist anymore. Everything works like a charm now.

**Comment by ShadowWhisperer** [comment]
@alexvrv 

Issue for me is, those files do exist. If I remove them, the errors go away. If I move them back, errors return. Those 3 movies show in my movie list and play fine.

**Comment by hugalafutro** [comment]
> [@alexvrv](https://github.com/alexvrv)
> 
> Issue for me is, those files do exist. If I remove them, the errors go away. If I move them back, errors return. Those 3 movies show in my movie list and play fine.

^^ This is the gist of the issue for me as well. The files exist, play, are indexed, transcode whatever have you perfectly, just show up this warning in log, going on for 6 years at this point.

**Comment by ShadowWhisperer** [comment]
@hugalafutro 

Renaming the video files (added the resolution to the name) seemed to do it. Errors no longer show in the log, but does not fix whatever the underlining issue is.

*Hoped this would fix my issue with some Collections not showing in the movie list, but it did not

**Comment by hugalafutro** [comment]
@ShadowWhisperer I cannot rename the files manually they are maintained by sonarr/radarr/lidarr container which rename them when they update the file due to higher quality or whatnot.

**Comment by Damocles-fr** [comment]
Still happening on latest version 10.11.6

**Comment by EricDuminil** [comment]
It still happens on 10.11.7.

If I understood correctly: the "LinkedAlternateVersions" value inside the JSON inside the DATA column of the BaseItems table still contains paths that have been since deleted.

WARNING: Be sure to make a backup of your jellyfin.db before running the queries!!!

Here's a small script to find links to STALE_PATH:

```sh
#!/bin/sh

# Find stale LinkedAlternateVersions in Jellyfin's DB.

STALE_PATH="/some/path/which/has/since/been/deleted/%"

QUERY="
SELECT
    b.Name,
    b.Id,
    j.value ->> 'Path'   AS LinkedPath,
    j.value ->> 'ItemId' AS LinkedItemId
FROM
    BaseItems b,
    json_each(json_extract(b.Data, '$.LinkedAlternateVersions')) AS j
WHERE
    b.Data IS NOT NULL
    AND json_array_length(json_extract(b.Data, '$.LinkedAlternateVersions')) > 0
    AND (j.value ->> 'Path') LIKE '${STALE_PATH}'
ORDER BY
    b.Name;
"
sqlite3 jellyfin.db "$QUERY"

## Or if you want to run the query on a mounted volume:
# docker run --rm --user root \
#     --entrypoint sqlite3 \
#     -v jellyfin-config:/config \
#     keinos/sqlite3 \
#     -cmd "PRAGMA temp_store=MEMORY;" \
#     -readonly /config/data/jellyfin.db \
#     "$QUERY"
```

And here's a script to remove the entries still linked to STALE_PATH.

Power Jellyfin down, make a backup of your config & DB, run this script, start Jellyfin again.

```sh
#!/bin/sh
set -e

# Remove stale LinkedAlternateVersions from Jellyfin's DB.
#
# Stop Jellyfin before running this script.
# A backup is created automatically before any changes.

STALE_PATH="/some/path/which/has/since/been/deleted/%"

UPDATE_CMD="
UPDATE BaseItems
SET Data = json_set(Data, '$.LinkedAlternateVersions',
    (SELECT json_group_array(j.value)
     FROM json_each(json_extract(Data, '$.LinkedAlternateVersions')) AS j
     WHERE (j.value ->> 'Path') NOT LIKE '${STALE_PATH}'))
WHERE Data LIKE '%${STALE_PATH}'
  AND json_array_length(json_extract(Data, '$.LinkedAlternateVersions')) > 0;

SELECT changes() || ' row(s) updated.';
"

echo "Backing up database..."
docker run --rm --user root \
    --entrypoint sqlite3 \
    -v jellyfin-config:/config \
    keinos/sqlite3 \
    -cmd "PRAGMA temp_store=MEMORY;" \
    -cmd ".backup /config/data/jellyfin.db.bak" \
    /config/data/jellyfin.db \
    ""

echo "Removing stale links matching: ${STALE_PATH}"
docker run --rm --user root \
    --entrypoint sqlite3 \
    -v jellyfin-config:/config \
    keinos/sqlite3 \
    -cmd "PRAGMA temp_store=MEMORY;" \
    /config/data/jellyfin.db \
    "$UPDATE_CMD"

```

Duplicate entries in my library are now finally gone, and I don't see any error in the logs anymore.

**Comment by daniko** [comment]
> It still happens on 10.11.7.
> Duplicate entries in my library are now finally gone, and I don't see any error in the logs anymore.

Amazing, this worked for me and the queries also provided extra items that did not appear on the logs that were stale as well. Just had to modify some commands as they were not compatible with the jellyfin sqlite:

For stale query:
```
#!/bin/sh

# Find stale LinkedAlternateVersions in Jellyfin's DB.

STALE_PATH="/some/path/which/has/since/been/deleted/%"

QUERY="
SELECT
    b.Name,
    b.Id,
    json_extract(j.value, '$.Path')   AS LinkedPath,
    json_extract(j.value, '$.ItemId') AS LinkedItemId
FROM
    BaseItems b,
    json_each(json_extract(b.Data, '$.LinkedAlternateVersions')) AS j
WHERE
    b.Data IS NOT NULL
    AND json_array_length(json_extract(b.Data, '$.LinkedAlternateVersions')) > 0
    AND json_extract(j.value, '$.Path') LIKE '${STALE_PATH}'
ORDER BY
    b.Name;
"

sqlite3 jellyfin.db "$QUERY"
```

For stale cleanup:
```
#!/bin/sh
set -e

# Remove stale LinkedAlternateVersions from Jellyfin's DB.
#
# Stop Jellyfin AND BACKUP jellyfin.db before running this script!!!

STALE_PATH="/some/path/which/has/since/been/deleted/%"

UPDATE_CMD="
UPDATE BaseItems
SET Data = json_set(Data, '$.LinkedAlternateVersions',
    (SELECT json_group_array(j.value)
     FROM json_each(json_extract(Data, '$.LinkedAlternateVersions')) AS j
     WHERE json_extract(j.value, '$.Path') NOT LIKE '${STALE_PATH}'))
WHERE Data LIKE '%${STALE_PATH}'
  AND json_array_length(json_extract(Data, '$.LinkedAlternateVersions')) > 0;

SELECT changes() || ' row(s) updated.';
"

#cp jellyfin.db jellyfin.db.bak.stale
sqlite3 jellyfin.db "$UPDATE_CMD"
```

Thanks a lot!

**Comment by eharris** [comment]
This is still happening for me in 10.11.11.  Please reopen.

**Comment by ShadowWhisperer** [comment]
@eharris 

Patched version hasn't been released yet.

> There should be fixes in the upcoming 12.0 release that address this issue.

https://github.com/jellyfin/jellyfin/issues/14106

**Comment by daniko** [comment]
He might want to test 12.0 rc4 maybe?
